manticore/ethereum/plugins.py

```diff
@@ -142,3 +142,73 @@ class VerboseTraceStdout(Plugin):
 
     def will_evm_execute_instruction_callback(self, state, instruction, arguments):
         print(state.platform.current_vm)
+
+
+class KeepOnlyIfStorageChanges(Plugin):
+    """ TODO: reword
+        This state will ignore states that are the result of executing a
+        transaction that did not write to the storage.
+
+        When this plugin is enabled transactions that wont write to the storage
+        are considered not to change the evm world state and hence ignored as a
+        starting point for the following human transaction.
+    """
+
+    def did_open_transaction_callback(self, state, tx, *args):
+        """ We need a stack. Each tx (internal or not) starts with a "False" flag
+            denoting that it did not write anything to the storage
+        """
+        state.context["written"].append(False)
```

Comment:
Hmmm... how does it differ from `self.locked_context(string) as ctx`? That would be a global state and here we use a context local/tied to the state?